### ISSUE TITLE
[Feature request] Possible to MailMerge

### ISSUE THREAD

**Issue by esmoyer** [Stirling-Tools/Stirling-PDF#237]
I just got this spun up and exploring around but was curious is it possible or do would it ever be possible to do any type of "mail merge" by taking a CSV with data and mapping the columns to fields in the PDF to then spit out multiple PDFs? Similar to this: https://evermap.com/automailmerge.asp. Just curious. Either way, amazing application; can't wait to dig in more!

**Comment by Frooodle** [comment]
Seems quite complex, I'll add it to the list :) 
But focusing on other features for now

**Comment by esmoyer** [comment]
No worries, just wanted to be sure I didn't overlook something if it was already there since you have already done so much! Appreciate it!

**Comment by esmoyer** [comment]
Just curious if any other thought has been given to this? Thanks!

**Comment by Frooodle** [comment]
@esmoyer I am wanting to begin this work, could you provide a example PDF and csv file i can test on and expected output?

**Comment by esmoyer** [comment]
> @esmoyer I am wanting to begin this work, could you provide a example PDF and csv file i can test on and expected output?

Absolutely, here is a really simple example. The CSV could have other random columns, but in this case the PDF to be generated only need the 3 data points. Not sure what would be easier to implement (or if it makes a difference on how concatenation works as it could technically be done in the CSV prior to upload). But basically a PDF form with the fields to be mapped/populated would be uploaded then upload the CSV with the data and then map the fields across. Ideally, it would be nice to be able to auto-name the generated PDF based on a field or fields from the CSV, but then it would spit out an individual PDF for each line in the CSV. Let me know if you need further examples or clarifications!

[12w8e.pdf](https://github.com/Stirling-Tools/Stirling-PDF/files/14749753/12w8e.pdf)
[2nuef.pdf](https://github.com/Stirling-Tools/Stirling-PDF/files/14749754/2nuef.pdf)
[16xia.pdf](https://github.com/Stirling-Tools/Stirling-PDF/files/14749755/16xia.pdf)
[mailmerge_example_csv.csv](https://github.com/Stirling-Tools/Stirling-PDF/files/14749756/mailmerge_example_csv.csv)

**Comment by balazs-szucs** [comment]
Duplicate of #3569 

TODO; close with PR with #3569

**Comment by balazs-szucs** [comment]
I prefer JSON, so it'll be with JSON. (Sorry; a possible improvement for the future could be support for CSV or other similar formats.)

I created a minimal working example, and it is definitely possible. Backend-wise, it's not very complicated; frontend-wise, it is rather complicated sadly, since you generate fields on the frontend, as well as a JSON sample.

Basically, I want the frontend to provide a minimal UI to fill the form there and have, at the bottom, an auto-generated JSON they can fill out and use as a sample for automation.

**Comment by balazs-szucs** [comment]
You can expect something like this:

```
[
  {
    "Address_1": "",
    "Address_2": "",
    "Birthdate": "",
    "City": "",
    "Emergency_Contact": "",
    "Emergency_Phone": "",
    "Emergency_Relationship": "",
    "Name_First": "",
    "Name_Last": "",
    "Name_Middle": "",
    "Name_Prefix": "",
    "Name_Suffix": "",
    "SSN": "",
    "STATE": "",
    "Telephone_Home": "",
    "Telephone_Work": "",
    "ZIP": "",
    "ASSOCIATES DEGREE": false,
    "BACHELORS DEGREE": false,
    "COLLEGE NO DEGREE": false,
    "HIGH SCHOOL DIPLOMA": false,
    "MASTERS DEGREE": false,
    "OTHER DOCTORATE": false,
    "PHD": false,
    "PROFESSIONAL DEGREE": false,
    "Sex": false,
    "TRADE CERTIFICATE": false
  }
]
```

An JSON representation of; 

[interactiveform_enabled.pdf](https://github.com/user-attachments/files/22562963/interactiveform_enabled.pdf)
